### PR TITLE
KAFKA-10393: messages for fetch snapshot and fetch

### PR DESCRIPTION
**Pull request by jsancio**
Adds the changes to Fetch and FetchSnapshot as describe in KIP-630.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/resources/common/message/FetchSnapshotRequest.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      "about": "The broker ID of the follower." },
+    { "name": "MaxBytes", "type": "int32", "versions": "0+",
+      "about": "The maximum bytes to fetch from all of the snapshots." },
+    { "name": "Topics", "type": "[]TopicSnapshot", "versions": "0+",
+      "about": "The topics to fetch.", "fields": [
+      { "name": "Name", "type": "string", "versions": "0+", "entityType": "topicName",
+        "about": "The name of the topic to fetch." },
+      { "name": "Partitions", "type": "[]PartitionSnapshot", "versions": "0+",
+        "about": "The partitions to fetch.", "fields": [
+        { "name": "Index", "type": "int32", "versions": "0+",
```
Comment (chia7712) [inline clients/src/main/resources/common/message/FetchSnapshotRequest.json:33]: it seems to me ```PartitionIndex``` is a more common name in protocol.

File: clients/src/main/resources/common/message/FetchSnapshotResponse.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      "about": "The duration in milliseconds for which the request was throttled due to a quota violation, or zero if the request did not violate any quota." },
+    { "name": "ErrorCode", "type": "int16", "versions": "0+", "ignorable": false,
+      "about": "The top level response error code." },
+    { "name": "Topics", "type": "[]TopicSnapshot", "versions": "0+",
+      "about": "The topics to fetch.", "fields": [
+      { "name": "Name", "type": "string", "versions": "0+", "entityType": "topicName",
+        "about": "The name of the topic to fetch." },
+      { "name": "Partitions", "type": "[]PartitionSnapshot", "versions": "0+",
+        "about": "The partitions to fetch.", "fields": [
+        { "name": "Index", "type": "int32", "versions": "0+",
```
Comment (chia7712) [inline clients/src/main/resources/common/message/FetchSnapshotResponse.json:33]: ditto

**Comment by jsancio** [comment]
Thanks for the review @chia7712 and sorry for the delay but I have decided to merge this PR with https://github.com/apache/kafka/pull/9553.

I will take care of your comments in that PR!

File: clients/src/main/resources/common/message/FetchSnapshotRequest.json (hunk 1 above)
Comment (jsancio) [reply clients/src/main/resources/common/message/FetchSnapshotRequest.json:33]: FYI, @chia7712 I updated this field in https://github.com/apache/kafka/pull/9553 to `Partition` to match Fetch version.

File: clients/src/main/resources/common/message/FetchSnapshotResponse.json (hunk 2 above)
Comment (jsancio) [reply clients/src/main/resources/common/message/FetchSnapshotResponse.json:33]: FYI, @chia7712 I updated this field in https://github.com/apache/kafka/pull/9553 to `Partition` to match Fetch version.

**Pull request closed without merging**